### DISCUSSION TITLE
Sprint #3  - Link to User stories , Important tasks and hight level issues

### DISCUSSION THREAD

**Discussion by DominiqueProulx** [Broke-Brokers/soen341_fproject2023#130] [General]
[Sprint 3 Guidelines.docx.pdf](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/files/13246370/Sprint.3.Guidelines.docx.pdf)

[Sprint 3 wiki ](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/wiki/Sprint-3)
## Sprint 3

top level user story 
-1 : HOMEBUYERS #24 
-2 : RENTER #29 
-3 : BROKERS #34 
-4 : SYSTEM ADMINISTRATOR #37 

## 3. set up automatic unit tests 

- [x]  Set up automatic unit test for every new file created 

## US-1.4 Homebuyers Search for brokers
***Issue for User story*** 
#131 
***Issues for main tasks*** 

- [x] TASK - 1.4.1 Mockup for search for brokers page #132 
- [x] TASK - 1.4.2 Add the search for broker  link to homebuyer and guest navigation bar #133 
- [x] TASK - 1.4.3 Create Search-for-Broker.js   page and route from App.js #134 
- [x] TASK - 1.4.5 Search-for-brokers page  grid  gallery of brokers #137
- [x] TASK - 1.4.6 Search for brokers Search bar and filters #138
- [x] TASK - 1.4.7 Add brokers fields to   brokers database #139
- [x] TASK - 1.4.8 contact brokers form #140
- [x] TASK - 1.4.9 brokers contact email sent + confirmation email to user #141
- [x] TASK - 1.4.12 Create Brokers-Client.js page and route from App.js #159
- [x] TASK - 1.4.10 Brokers CRUD  to clients #158 
- [ ] TASK - 1.4.11 All database links done for 1.4 #157 

***Issue for acceptance test*** 
#285 
#283 

--- 
## US-2.4 Renters Search for brokers

#135 
tasks are the same as 1.4. 

---
## US- 3.1 Submit Offers: Brokers can submit offers for buying properties on behalf of their clients, and track the status of their offers.

***Issue for User story*** 

#143 
***Issues for main tasks*** 

- [x] TASK - 3.1.1 Mockup for submit offer feature #144
- [x] TASK - 3.1.2 Brokers navigation Bar  #145
- [x] TASK - 3.1.3 Make an Offer on this house button in property page #146
- [x] TASK - 3.1.4 Offer on property form #147
- [ ] TASK - 3.1.5 Make offer on house confirmation emails #148
- [x] TASK - 3.1.6 Update "offer sent" section of broker "offer" page #149
- [ ] TASK - 3.1.7 All database links done for 3.1 #160 

***Issue for acceptance test*** 
#287 
#289 

## US- 3.5 Offer Management

Brokers can review and manage offers submitted by home buyers for properties they represent Offer Management: Brokers can review and manage offers submitted by home buyers for properties they represent

***Issue for User story*** 
#142 
***Issues for main tasks*** 
- [x] TASK - 3.5.1 Mockup for Offer Management #150 
- [x] TASK - 3.5.3 Create Broker- offers.js page and route from App.js #152 
- [x] TASK - 3.5.4 Broker-offers page "offer received" section #153 
- [x] TASK - 3.5.5 Broker-offers page "offer sent" section #154 
- [x] TASK - 3.5.6 Broker-offers page "past received" and "past sent" sections #155 
- [x] TASK - 3.5.7 Emails  : email client on offer received + automatic email after accept or rejet offer or negociation #156 
- [ ] TASK - 3.5.8 All database links done for 3.5 #161 

***Issue for acceptance test***
#290 
#291 

### 0 level TASKS

- [x] 0.5.1 Navigation per user type #220

- [ ] Make login work  -- see cookie added as part of #220
- [ ] stage gituhub pages #236 

### Other important tasks to overtake

- [x] #208 write the acceptance tests for 1.4, 3.1 and 3.5
- [x] #162  meeting minutes for sprint 3
- [x] #163 sprint 4 backlog
- [x] #225 
- [ ] Resolve database/Firebase crash #298 

## Mockup
https://web.goodnotes.com/s/umHDEfCzU5jKA2j4qmA1Sq#page-12

**Comment by DominiqueProulx** [comment]
I have created an issue on meetings for sprint 3 and end of sprint 2.

Sprint 2 review, sprint 2 retrospect, sprint 3 planning :
https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/issues/207

**Comment by DominiqueProulx** [comment]
I'm picking up #134- 1.4.3 and #137- 1.4.5 to get started on it

**Comment by isaisa002** [comment]
[BrokeBrokers_Sprint3_meeting2_2023-11-03.pdf](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/files/13247073/BrokeBrokers_Sprint3_meeting2_2023-11-03.pdf)
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/118073058/f0d8ff9e-14e3-4ca2-b693-49278d2fd348)
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/118073058/c877d961-51fe-4adf-a469-5b0bcef5aa58)
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/118073058/3d602505-c88d-4304-bdbe-e9ad9861cd5c)

**Reply by DominiqueProulx** [reply]
Perfect! Thanks @isaisa002 !

Only thing I notice :
Sprint 2 -  the logical test on  PROFIE is missing (to show CRUD on properties or  CRUD on brokers depending on suerType)

**Reply by isaisa002** [reply]
Added it as a task in login page second bullet point 

- Isabelle & Taylor ... in profile.js

**Reply by DominiqueProulx** [reply]
@taylcr -For :  [ ] Make login work and update cookie for usertypeID   I did not create an issue, it's yours to create. 

I assumed we are using cookies once user is logged in to keep track of usertypeID ? 
https://www.npmjs.com/package/js-cookie

I will base the navigation bar logic on that value(Isabelle provided the database field name) "usertypeID"

**Comment by DominiqueProulx** [comment]
@taylcr 
> @taylcr -For : [ ] Make login work and update cookie for usertypeID I did not create an issue, it's yours to create.

did you create an issue for this? I need to leave you comments. We are missing a logout page or button , how do you want to handle this? 

thanks

**Comment by DominiqueProulx** [comment]
Hello! @Broke-Brokers/broke-brokers  any updates on where we are at in these issues? 
Even if it's not done please update on progress

**Reply by isaisa002** [reply]
### Crud on broker by admin
Update still can't work see #37 for detailed progress 
### populating database with updated fields
- completed for broker, offers,client
- In progress for home buyer, renter 
 
### Client page for broker
- Didn't begin to prioritize Offer page for US 3.5

### Search for broker
- Can search a broker according to its language speak & years of experience

**Reply by JPEUMZZ** [reply]
### Search for property
- Can research by city, for sale, for rent, within price range, number of bathrooms & bedrooms
- need to push my update on the main but it's on my branch 
- UI needs enhancement but not a priority now
- Result gallery not connected to property page of the property clicked 

### Search for brokers 
- Can search a broker by its name, language speak & years of experience 
- UI needs enhancement but not a priority 

### Make an offer 
- Page created using the "sell" page, will change name after 
- Form in process

**Reply by DominiqueProulx** [reply]
Thanks! let's try to comment directly in the issues as we go yes?

**Reply by DominiqueProulx** [reply]
@isaisa002  Thanks for the update!!

> ### Crud on broker by admin
> Update still can't work see #37 for detailed progress

thanks

> ### populating database with updated fields
> * completed for broker, offers,client
> * In progress for home buyer, renter

Please update in #157 , #160 
Backlog  had Rebecca assigned on these tasks  - Do I change it ? 

> ### Client page for broker
> * Didn't begin to prioritize Offer page for US 3.5

Ok we'll see if I / somebody can do a quick something later 

> ### Search for broker
> * Can search a broker according to its language speak & years of experience

Please update in #138

---
@JPEUMZZ Awesome Thank you for the update!! 

> ### Search for property
> * Can research by city, for sale, for rent, within price range, number of bathrooms & bedrooms

Woot !! awesome
Please update in #96 

> * need to push my update on the main but it's on my branch

please do it fast, we need to make sure we have no problem in integration pipeline and we need to do acceptance test (can we do it with yesterday's merge?)
 We also  need to create unit tests -- please see comments and see discussion on testing  


> * UI needs enhancement but not a priority now
> * Result gallery not connected to property page of the property clicked

Result gallery is connected to property page in the main branch , your branch is too old to see it maybe? 


> ### Search for brokers
> * Can search a broker by its name, language speak & years of experience
> * UI needs enhancement but not a priority

Please comment on issue #138 
Please merge change so we can write unit test and do acceptance test on that page.

> ### Make an offer
> * Page created using the "sell" page, will change name after

? the route and page were createwd as part of #152. 

> * Form in process

Please sent updates and screenshots on #147

**Reply by JPEUMZZ** [reply]
Jtest doesn't create automatically unit tests for us every time we tried to push or merge something
